src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,219 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get the car by id " + id, e);
+        }
+
+        if (car != null) {
+            car.setDrivers(getAllDriversAssignedCar(car.getId()));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String queryGetAllCars =
+                "SELECT cars.id AS id, model, manufacturer_id, name, country FROM cars "
+                + "JOIN manufacturers ON cars.manufacturer_id = manufacturers.id"
+                + " WHERE cars.is_deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement allCarsStatement = connection.prepareStatement(queryGetAllCars)) {
+            ResultSet carsResultSet = allCarsStatement.executeQuery();
+            while (carsResultSet.next()) {
+                cars.add(getCar(carsResultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get list of cars from the DB", e);
+        }
+
+        setDriversForCars(cars);
+        return cars;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String queryAllCarsByDriver =
+                "SELECT cars.id, model, manufacturer_id, manufacturers.name, manufacturers.country "
+                + "FROM cars JOIN cars_drivers ON cars.id = id_car JOIN manufacturers "
+                + "ON manufacturer_id = manufacturers.id "
+                + "WHERE id_driver = ? AND cars.is_deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement allCarsByDriverStatement =
+                        connection.prepareStatement(queryAllCarsByDriver)) {
+            allCarsByDriverStatement.setLong(1, driverId);
+            ResultSet carsResultSet = allCarsByDriverStatement.executeQuery();
+            while (carsResultSet.next()) {
+                cars.add(getCar(carsResultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars by the driver with id "
+                    + driverId, e);
+        }
+
+        setDriversForCars(cars);
+        return cars;
+    }
+
+    private void assignDriversToTheCar(Car car) {
+        String query = "INSERT INTO cars_drivers (id_car, id_driver) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Driver driver: car.getDrivers()) {
+                try (PreparedStatement createCarsDriversStatement =
+                             connection.prepareStatement(query)) {
+                    createCarsDriversStatement.setLong(1, car.getId());
+                    createCarsDriversStatement.setLong(2, driver.getId());
+                    createCarsDriversStatement.executeUpdate();
```

Comment:
you can use only one try-with-resources, change only `id_driver` each iteration, and you can use for-loop after defineding `PreparedStatement`